adr/query.py

```diff
@@ -142,19 +132,38 @@ def run_query(name, config, args):
     return query_activedata(query_str, config.url)
 
 
-def format_query(query, config):
+def format_query(query, config, remainder=[]):
     """Takes the output of the ActiveData query and performs formatting.
 
     The result(s) from a query call to ActiveData is returned,
     which is then formatted as per the fmt argument.
 
     :param name query: name of the query file to be run.
+    :param remainder: user contexts
     :param Configuration config: config object.
     """
     if isinstance(config.fmt, str):
         fmt = all_formatters[config.fmt]
 
-    result = run_query(query, config, Namespace(**FAKE_CONTEXT))
+    # change list to dict
+    args = {}
+    i = 0
+    while i < len(remainder):
+        args[remainder[i]] = remainder[i+1]
+        i += 2
```

Comment:
I don't think this is going to work in all cases. Not all arugments are guaranteed to have a value. And some might even be positional. For example remainder could look like this:

    ["--foo", "bar",  "--colour=blue", "positional"]

It's not safe to assume that every second argument is going to be a value. Just given that list, it's impossible to tell if "--foo" is an argument with value "bar" or if "--foo" is a bool and "bar" is positional. If the user specifies "--foo=bar", that'll also show up as a single item in this list. The only way to do this properly is to try and build a new ArgumentParser object based on the query context, and then parse the remainder using that.

p.s I've been thinking about this a little bit, and I think we need to invent a new format for defining context values that is easy to apply to both `argparse` and the webapp. Maybe we can talk about  this this week.